sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/processor/SimpleSpanProcessor.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace
+package processor
+
+import cats.MonadThrow
+import cats.effect.std.Console
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+import org.typelevel.otel4s.sdk.trace.exporter.SpanExporter
+import org.typelevel.otel4s.trace.SpanContext
+
+import scala.util.control.NonFatal
+
+/** An implementation of the [[SpanProcessor]] that passes ended
+  * [[data.SpanData SpanData]] directly to the configured exporter.
+  *
+  * @note
+  *   this processor exports spans individually upon completion, resulting in a
+  *   single span per export request.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#simple-processor]]
+  *
+  * @tparam F
+  *   the higher-kinded type of a polymorphic effect
+  */
+final class SimpleSpanProcessor[F[_]: MonadThrow: Console] private (
+    exporter: SpanExporter[F],
+    exportOnlySampled: Boolean
+) extends SpanProcessor[F] {
+
+  val isStartRequired: Boolean = false
+  val isEndRequired: Boolean = true
+
+  def onStart(parentContext: Option[SpanContext], span: SpanRef[F]): F[Unit] =
+    MonadThrow[F].unit
+
+  def onEnd(span: SpanData): F[Unit] = {
+    val canExport = !exportOnlySampled || span.spanContext.isSampled
+    doExport(span).whenA(canExport)
+  }
+
+  private def doExport(span: SpanData): F[Unit] =
+    exporter.exportSpans(List(span)).onError {
+      case e if NonFatal(e) =>
```

Comment:
You never need non-fatal checks within the effect, if a non-fatal error has been countered the JVM has already terminated :grin:

Also `onError` just-side-channels, so the error continues being raised. Wouldn't this possibly lead to duplicate error reporting?